src/common/msb.rs

```diff
@@ -1,26 +1,36 @@
+#[derive(Debug, Eq, PartialEq)]
+pub enum Bit {
+    _0 = 0,
+    _1 = 1,
+}
+
 trait GetBit {
-    fn get_bit(&self, bit_index: usize) -> Option<u8>;
+    fn get_bit(&self, bit_index: usize) -> Option<Bit>;
 }
 
 impl GetBit for u8 {
-    fn get_bit(&self, bit_index: usize) -> Option<u8> {
+    fn get_bit(&self, bit_index: usize) -> Option<Bit> {
         if bit_index < 8 {
-            let shift = 1 << (7 - bit_index);
-            let bit = (self & shift) != 0;
-            Some(bit as u8)
+            let mask = 1 << (7 - bit_index);
+            let bit = ((self & mask) != 0) as u8;
+            match bit {
```

Comment:
It might be nice to implement a `From` trait for `Bit` on `u8` so instead of `match` statements everywhere we can just do `Some(Bit::from(bit))`

Reply:
Also FWIW would it make more sense to calculate `!(self & mask) & 1` to avoid the u8 to boolean to u8 type conversion?

Reply:
Do you mean from `Bit` to `u8`? Or from `u8` to `Bit`? 

In the case of `Bit` to `u8`, that is already given freely, and we can do:
```
let bit = Bit::_0;
let val = bit as u8; // or u16, u32, etc.
```

In the case of `u8` to `Bit`, I wouldn't suggest implementing that (i.e. `impl From<u8> for Bit`) because of the possible 253 values of `u8` that are invalid; it would have to be a `impl TryFrom<u8> for Bit`, which would return an `Option<Bit>`. 

The reason I can safely use match on a `u8` in `get_bit(..)` is because I know that the value of the `u8` is always 0 or 1. In the wild, with a `From` trait, I can't guarantee that.

Reply:
I think you're right that I can simplify the actual calculation of bit here. Will change!

Reply:
Oh I see, I hadn't realized that `Bits` could all ready be used as a `u8` literal